solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -474,7 +539,33 @@ contract InterchainAccountRouter is Router {
     ) private returns (bytes32) {
         require(_router != bytes32(0), "no router specified for destination");
         emit RemoteCallDispatched(_destination, msg.sender, _router, _ism);
-        return mailbox.dispatch(_destination, _router, _body);
+        return mailbox.dispatch{value: msg.value}(_destination, _router, _body);
+    }
+
+    /**
+     * @notice Dispatches an InterchainAccountMessage to the remote router with hook metadata
+     * @param _destination The remote domain
+     * @param _router The address of the remote InterchainAccountRouter
+     * @param _ism The address of the remote ISM
+     * @param _body The InterchainAccountMessage body
+     * @param _hookMetadata The hook metadata to override with for the hook set by the owner
+     */
+    function _dispatchMessageWithMetadata(
+        uint32 _destination,
+        bytes32 _router,
+        bytes32 _ism,
+        bytes memory _body,
+        bytes memory _hookMetadata
+    ) private returns (bytes32) {
+        require(_router != bytes32(0), "no router specified for destination");
+        emit RemoteCallDispatched(_destination, msg.sender, _router, _ism);
```

Comment:
I would prefer to have only one `mailbox.dispatch` and `emit RemoteCallDispatched` in the contract
can we refactor slightly?

Reply:
I cannot put empty metadata so we need an if else regardless of having just one function call mailbox.dispatch